wgpu-hal/src/gles/command.rs

```diff
@@ -291,9 +291,13 @@ impl crate::CommandEncoder for super::CommandEncoder {
             if !bar.usage.from.contains(wgt::BufferUses::STORAGE_READ_WRITE) {
                 continue;
             }
+            let buffer = match &bar.buffer.backing {
+                &BufferBacking::Gl { raw } | &BufferBacking::GlCachedOnHost { raw, .. } => raw,
+                BufferBacking::Host { .. } => unreachable!(),
+            };
```

Comment:
```suggestion
            let buffer = match bar.buffer.backing {
                BufferBacking::Gl { raw } | &BufferBacking::GlCachedOnHost { raw, .. } => raw,
                BufferBacking::Host { .. } => unreachable!(),
            };
```

This feels like a weird reference-dereference? If it needs a ref, could we use `ref raw`

Reply:
Also in general I don't like taking references in match expressions, preferring to explictly using `ref`/`ref mut`. This is a slightly less conservative version of the clippy lint against match ergonomics.